### ISSUE TITLE
Multi client capability

### ISSUE THREAD

**Issue by ranjanpr** [HTD/Woof.IPC#1]
First of all kudos for making a good library for IPC. I have used this as named server in some of my projects.
However i am unable to do ipc between a server and multiple clients scenario. 
Is it possible to support multi client connections to same server using this library simultaneously?

**Comment by HTD** [comment]
Yes, I have a production code that uses that exact scenario - one server (Windows service running as System user) and multiple clients (UI applications running on user accounts). What is happening? Do you get an error message? I tested the module with multiple clients and it seemed to work.

**Comment by ranjanpr** [comment]
I was able to do the connection but only with the client written using same library. However my c# application with IpcServer needs to connect with IPC clients using named pipe written in c++. One thing which i saw is IpcServer creates named pipe with suffixes `-In` and `-Out` attached to it. Is it intended?

**Comment by HTD** [comment]
Pipe names are intended. Is something wrong with them? The access violation could probably be fixed with a registry hack, however, I forgot how I resolved a similar issue earlier. It may be hiding in one of my work projects sources somewhere. At work the lib was used to communicate the user-level UI with system-level service. So the user didn't see UAC prompt each time a service setting is changed. Besides - the system (service) user has no direct access to user's UI so I found no other way than IPC. The access issue emerged when I needed to send data to a service from System Installer user. Currently I'm totally overloaded with work, so I'll be able to get back to it in later.

**Comment by ranjanpr** [comment]
Thanks for the response @HTD . I had to discard this library and use naked `NamedPipeStream` for my use case where i needed to read-write simultaneously with multiple clients. As far as pipename is concerned, it would be preferable to not add any suffixes. I was also able to resolve the access violation issue by passing in security parameter while creating `NamedPipeStream` like this

```
PipeSecurity security = new PipeSecurity();
            security.AddAccessRule(new PipeAccessRule($"{Environment.UserDomainName}\\{Environment.UserName}", PipeAccessRights.ReadWrite, System.Security.AccessControl.AccessControlType.Allow));

            namedPipeServer = new NamedPipeServerStream(PipeName, PipeDirection.InOut, 1, PipeTransmissionMode.Message, PipeOptions.Asynchronous, 65536, 65536, security);
```

**Comment by HTD** [comment]
Thanks for sharing your solution. It might become handy one day.
